http2/src/main/java/com/blazemeter/jmeter/http2/sampler/HTTP2StreamHandler.java

```diff
@@ -12,403 +21,405 @@
+      getPropDefault("httpsampler.ignore_failed_embedded_resources", false);
+
+
+  private final CompletableFuture<Void> completedFuture = new CompletableFuture<>();
+  private HTTP2SampleResult result;
+  private HTTP2Connection parent;
+  private byte[] responseBytes;
+  private HeaderManager headerManager;
+  private CookieManager cookieManager;
+  private boolean first = true;
+  private int timeout = 0;
+
+  static {
+    getParsers();
+  }
+
+  private static void getParsers() {
+    String[] parsers = JOrphanUtils.split(RESPONSE_PARSERS, " ", true);
+    for (final String parser : parsers) {
+      String classname = JMeterUtils.getProperty(parser + ".className");
+      if (classname == null) {
+        LOG.error(
+            "Cannot find .className property for " + parser + ", ensure you set property:'" + parser
+                + ".className'");
+        continue;
+      }
+      String typeList = JMeterUtils.getProperty(parser + ".types");
+      if (typeList != null) {
+        String[] types = JOrphanUtils.split(typeList, " ", true);
+        for (final String type : types) {
+          LOG.info("Parser for " + type + " is " + classname);
+          PARSERS_FOR_CONTENT_TYPE.put(type, classname);
         }
+      } else {
+        LOG.warn("Cannot find .types property for " + parser
+            + ", as a consequence parser will not be used, to make it usable, define property:'"
+            + parser + ".types'");
+      }
     }
-
-    private final CompletableFuture<Void> completedFuture = new CompletableFuture<>();
-    private HTTP2SampleResult result;
-    private HTTP2Connection parent;
-    private URL url;
-    private byte[] responseBytes;
-    private HeaderManager headerManager;
-    private CookieManager cookieManager;
-    private boolean first = true;
-    private int timeout = 0;
-
-    public HTTP2StreamHandler(HTTP2Connection parent, URL url, HeaderManager headerManager, CookieManager cookieManager, HTTP2SampleResult sampleResult) {
-        this.result = sampleResult;
-        this.parent = parent;
-        this.url = url;
-        this.cookieManager = cookieManager;
-        this.headerManager = headerManager;
+  }
+
+  public HTTP2StreamHandler(HTTP2Connection parent, HeaderManager headerManager,
+      CookieManager cookieManager, HTTP2SampleResult sampleResult) {
+    this.result = sampleResult;
+    this.parent = parent;
+    this.cookieManager = cookieManager;
+    this.headerManager = headerManager;
+  }
+
+  public CompletableFuture<Void> getCompletedFuture() {
+    return completedFuture;
+  }
+
+  @Override
+  public Listener onPush(Stream stream, PushPromiseFrame frame) {
+    MetaData.Request requestMetadata = ((MetaData.Request) frame.getMetaData());
+
+    URL url = null;
+    try {
+      url = requestMetadata.getURI().toURI().toURL();
+    } catch (MalformedURLException | URISyntaxException e) {
+      LOG.error("Failed when parsed Push URL", e);
```

Comment:
How it will continue without URL?

Reply:
Here we are converting from Jetty's URI to Java's URI. This is only to set properties on the SampleResponse.